pkg/kubectl/sorting_printer.go

```diff
@@ -60,8 +61,28 @@ func (s *SortingPrinter) sortObj(obj runtime.Object) error {
 	if len(objs) == 0 {
 		return nil
 	}
+
+	if err := SortObjects(objs, s.SortField); err != nil {
+		return err
+	}
+
+	switch list := obj.(type) {
+	case *v1.List:
+		for ix := range objs {
+			data, err := v1.Codec.Encode(objs[ix])
```

Comment:
> This is a v1 List. a v1 list can contain things that aren't v1?

Yes.  `v1.List` is allowed to contain elements from other versions and should be allowed to contain things from other group/versions as well.